docs/develop/java/spring/rate-limiting/fixed-window/index-fixed-window-reactive-gears.mdx

```diff
@@ -0,0 +1,290 @@
+#### Function Registration
+
+1. At the bottom of the script, in the `# Function registration` section, we instantiate the
+[`GearsBuilder`(`GB`)](https://oss.redis.com/redisgears/runtime.html#gearsbuilder) using the `CommandReader`
+reader. The `GearsBuilder` "builds" the context of the function, in parameters, transformations, triggers, etc.
+2. We use the `map` method to performs a one-to-one mapping of records to the params of the `rate_limit`
+function via a mapper function callback.
+3. We can now invoke the `register` action to register the function as an event handler. The event in our case is the
+trigger `'RateLimiter'`.
+
+## RedisGears in SpringBoot
+
+In order to use our RedisGear function from our SpringBoot application we need to do a few things:
+
+1. Deploy the function to the Redis server
+2. Execute the function to get a yay/nay answer on each request
+
+## Lettuce Mod
+
+[LettuceMod](https://github.com/redis-developer/lettucemod) is a Java client for Redis Modules based on Lettuce created by [Julien Ruaux ](https://github.com/jruaux).
+It supports the following modules in standalone or cluster configurations:
+
+* RedisGears
+* RedisJSON
+* RediSearch
+* RedisTimeSeries
+
+To use LettuceMod we'll add the dependency to the Maven POM as shown:
+
+```xml
+<dependency>
+  <groupId>com.redis</groupId>
+  <artifactId>spring-lettucemod</artifactId>
+  <version>1.7.0</version>
+</dependency>
+```
+
+### Accessing Gears Commands in SpringBoot
+
+To access any of the LettuceMod supported modules we will inject a `StatefulRedisModulesConnection` in
+our `FixedWindowRateLimiterApplication` class as follows:
+
+```java
+@Autowired
+StatefulRedisModulesConnection<String, String> connection;
+```
+
+Add the matching import statement:
+
+```java
+import com.redis.lettucemod.api.StatefulRedisModulesConnection;
+```
+
+### Registering the Gears function
+
+We'll start by writing a function to determine whether the function with the trigger `RateLimiter` has been
+registered. It takes a `List` of `Registration`s and digs deep to extract the value of the `trigger` argument
+using the Java Streams API:
+
+```java
+private Optional<String> getGearsRegistrationIdForTrigger(List<Registration> registrations, String trigger) {
+  return registrations.stream().filter(r -> r.getData().getArgs().get("trigger").equals(trigger)).findFirst().map(Registration::getId);
+}
+```
+
+In the `@PostConstruct` annotated method `loadGearsScript` method:
+
+1. We retrieve an instance of the `RedisGearsCommands` from the previously injected `StatefulRedisModulesConnection`
+2. We get the currently registered Gears functions via the `dumpregistrations` method
+3. We pass the list of registrations to our `getGearsRegistrationIdForTrigger`
+4. If we don't find the registration we proceed to register the function:
+   - Load the function from the classpath into a `String` named `py`
+   - Use the `pyexecute` method passing the `py` script payload
+
+```java
+@PostConstruct
```

Comment:
This code comes up really bunched up in docasaurus, it could use some spacing and some line separation to make it a bit more readable